src/compile/data.ts

```diff
@@ -564,7 +667,16 @@ export namespace summary {
     return [];
   }
 
-  export function assemble(component: DataComponent): VgData[] {
+  export function parseLayer(model: LayerModel): SummaryComponent[] {
+    // we create new data sources for each summary in a layer
+    return [];
+  }
+
+  /**
+   * Assemble the summary. Needs a rename function because we cannot guarantee that the
+   * parent data before the children data.
+   */
+  export function assemble(component: DataComponent, rename: (name: string) => string): VgData[] {
```

Comment:
I also say that rename should happen in parse, not in assemble.  assemble should just read the name.  That's it.  `rename` parameter here should be removed.